### ISSUE TITLE
Promise never returns when using Pouch 5.0.0

### ISSUE THREAD

**Issue by Ampakinetic** [pouchdb/http-pouchdb#4]
It could be related to this issue?  https://github.com/nolanlawson/pouchdb-replication-stream/issues/41

**Comment by marten-de-vries** [comment]
Thanks for reporting.

If that's the case, using 'pouchdb/extras/promise' instead of 'pouchdb-promise' might help. Worth trying, I guess.

But could you give a full example of how to reproduce this?

**Comment by Ampakinetic** [comment]
Thank you, if I use:

var db = require('http-pouchdb')(PouchDB, 'http://localhost:51738/');

Then it never comes back from the put

But if I use:
var db = new PouchDB("http://localhost:51738/testing");

then it works, with the full code below:

```

        var db = new PouchDB("http://localhost:51738/testing");

        db.put({ _id: 'mydoc', "test": "yes" }).then(function (response) {

            db.allDocs({

                include_docs: true,

                attachments: true

            }).then(function (result) {

                res.send(JSON.stringify(result));

            }).catch(function (err) {

                console.log(err);

            });

        }).catch(function (err) {

            console.log(err);

        });
```

**Comment by Ampakinetic** [comment]
Thank you, if I use:

var db = require('http-pouchdb')(PouchDB, 'http://localhost:51738/');

Then it never comes back from the put

But if I use:
var db = new PouchDB("http://localhost:51738/testing");

then it works, with the full code below:

```
        var db = new PouchDB("http://localhost:51738/testing");

        db.put({ _id: 'mydoc', "test": "yes" }).then(function (response) {

            db.allDocs({

                include_docs: true,

                attachments: true

            }).then(function (result) {

                res.send(JSON.stringify(result));

            }).catch(function (err) {

                console.log(err);

            });

        }).catch(function (err) {

            console.log(err);

        });


```

**Comment by Ampakinetic** [comment]
Are you able to reproduce the issue as well or was I making a mistake?

**Comment by marten-de-vries** [comment]
Well, I can see that http-pouchdb is quite broken nowadays. See also #3. Seems like the whole concept needs to be rethought.

**Comment by marten-de-vries** [comment]
As soon as https://travis-ci.org/pouchdb/http-pouchdb/builds/86801127 passes, I'm releasing a new version that fixes this (tests pass locally, but I want that check).

**Comment by marten-de-vries** [comment]
Code for the fix is in 6180f32

**Comment by marten-de-vries** [comment]
```
marten@procyon:~/git/http-pouchdb$ npm publish
+ http-pouchdb@1.1.3
```
